rareapi/views/post.py

```diff
@@ -1,40 +1,62 @@
-"""View module for handling requests about posts"""
-
-from django.contrib.auth.models import User
+"""View module for handling requests about games"""
 from django.core.exceptions import ValidationError
 from rest_framework import status
 from django.http import HttpResponseServerError
 from rest_framework.viewsets import ViewSet
 from rest_framework.response import Response
 from rest_framework import serializers
 from rest_framework import status
-from rareapi.models import Post
-
+from rareapi.models import Post, RareUser
 
 
 class Posts(ViewSet):
-    """Post"""
 
     def list(self, request):
-        """Handle GET requests to post resource
-        Returns:
-            Response -- JSON serialized list of games
-        """
-        # Get all post records from the database
-        posts = Post.objects.all()
 
+        posts = Post.objects.all()
         user = request.auth.user
-
         if user is not None:
-            posts = posts.filter(user_id = user.id)
+            posts = posts.filter(user_id=user.id)
 
-        serializer = PostSerializer(
-            posts, many=True, context={'request': request})
+        user_id = self.request.query_params.get('user_id', None)
+        serializer = PostSerializer(posts, many=True, context={'request': request})
         return Response(serializer.data)
 
-class PostSerializer(serializers.ModelSerializer):
-    """ JSON serializer for posts"""
+    def retrieve(self, request, pk=None):
+        """Handle GET requests for single game
+        Returns:
+            Response -- JSON serialized game instance
+        """
+        try:
+            # `pk` is a parameter to this function, and
+            # Django parses it from the URL route parameter
+            #   http://localhost:8000/games/2
+            #
+            # The `2` at the end of the route becomes `pk`
+            post = Post.objects.get(pk=pk)
+            serializer = PostSerializer(post, context={'request': request})
+            return Response(serializer.data)
+        except Exception as ex:
+            return HttpResponseServerError(ex)
+
+   
+
+    
+
+
 
+
+"""Serializer for RareUser Info in a post"""         
+class PostRareUserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = RareUser
+        fields = ('id', 'bio', 'user')
+        depth = 1
+
+"""Basic Serializer for single post"""
+class PostSerializer(serializers.ModelSerializer):
+    user = PostRareUserSerializer(many=False)
     class Meta:
         model = Post
-        fields = ('id', 'user', 'category', 'title', 'publication_date', 'image_url')
\ No newline at end of file
+        fields = ('id', 'title', 'publication_date', 'content', 'user', 'category')
```

Comment:
this needs to include fields = ('id', 'user', 'category', 'title', 'publication_date', 'image_url', 'content', 'approved')